spec/web-server.spec.js

```diff
@@ -0,0 +1,259 @@
+describe("web-server", function () {
+    "use strict";
+
+    var fs, logger, restify, restMiddleware, webServer, WebServer;
+
+    WebServer = require("../lib/web-server");
+
+    beforeEach(function () {
+        fs = {
+            readFileSync: function () {}
+        };
+
+        logger = {
+            console: function () {},
+            debug: function () {},
+            error: function () {},
+            info: function () {},
+            warn: function () {}
+        };
+
+        restify = {
+            createServer: function () {
+                return {};
+            }
+        };
+
+        restMiddleware = function () {
+
+        };
+
+        spyOn(fs, "readFileSync");
+        spyOn(logger, "console");
+        spyOn(logger, "debug");
+        spyOn(logger, "error");
+        spyOn(logger, "info");
+        spyOn(logger, "warn");
+        spyOn(restify, "createServer").andReturn({});
+    });
+
+    beforeEach(function () {
+        webServer = new WebServer(fs, logger, restify, restMiddleware);
+    });
+
+    describe("constructor and configure", function () {
+        it("should set up properties without config options then calls configure", function () {
+            expect(webServer.config).toEqual({
```

Comment:
We should be testing the behavior, not the internal properties.  Mostly things like "did the instance get created" and "Is it set up for HTTPS automatically" or stuff.  We should not peek inside the webServer.config (as we discussed earlier today).